CRM/Core/Payment/ProcessorForm.php

```diff
@@ -75,7 +75,10 @@ public static function preProcess(&$form, $type = NULL, $mode = NULL) {
     $currency = CRM_Utils_Array::value('currency', $form->_values);
```

Comment:
It looks to me like this is where `$currency` should be stored by webform_civicrm.

Reply:
I saw this on a 4.7.25 system, then tested on a 4.7.29 (master).  It seems to be empty but currency is available via the URL parameters for the contribution page.

Reply:
My preference is to fix bugs related to webform_civicrm in the webform_civicrm module, or if it's a failing with the api, fix it on that level. This PR as it stands seems a little hackish.

Reply:
I'd really like to get standardised about where currency is stored - preferably add a 'getCurrency' function to the classes : CRM_Core_Form and CRM_Core_Payment and ensure that those functions get the right currency (the latter is for a related issue)